tests/performance/matrix_free_kokkos.cc

```diff
@@ -0,0 +1,331 @@
+std::vector<double>
+run(const unsigned int n_refinements)
+{
+  ConvergenceTable table;
+
+  const MPI_Comm comm = MPI_COMM_WORLD;
+
+  using Number     = double;
+  using VectorType = LinearAlgebra::distributed::Vector<Number, MemorySpace>;
+
+  const unsigned n_repetitions = 100;
+
+  parallel::distributed::Triangulation<dim> tria(comm);
+
+  GridGenerator::hyper_cube(tria);
+  tria.refine_global(n_refinements);
+
+  table.add_value("n_levels", tria.n_global_levels());
+  table.add_value("degree", degree);
+
+  table.add_value("n_cells", tria.n_global_active_cells());
+
+  const MappingQ1<dim> mapping;
+  const FE_Q<dim>      fe(degree);
+  const QGauss<1>      quadrature(degree + 1);
+
+  DoFHandler<dim> dof_handler(tria);
+  dof_handler.distribute_dofs(fe);
+
+  table.add_value("n_dofs", dof_handler.n_dofs());
+
+  AffineConstraints<Number> constraints;
+
+  std::chrono::time_point<std::chrono::system_clock> temp =
+    std::chrono::system_clock::now();
+  LaplaceOperator<dim, degree, Number, MemorySpace> laplace_operator(
+    mapping, dof_handler, constraints, quadrature);
+
+  const double dt_setup = std::chrono::duration_cast<std::chrono::nanoseconds>(
+                            std::chrono::system_clock::now() - temp)
+                            .count() /
+                          1e9;
+
+  VectorType src, dst;
+
+  laplace_operator.initialize_dof_vector(src);
+  laplace_operator.initialize_dof_vector(dst);
+
+  {
+    LinearAlgebra::distributed::Vector<Number> src_host(src.get_partitioner());
+
+    VectorTools::interpolate(dof_handler,
+                             AnalyticalFunction<dim, Number>(),
+                             src_host);
+
+    LinearAlgebra::ReadWriteVector<Number> rw_vector(
+      src.get_partitioner()->locally_owned_range());
+    rw_vector.import(src_host, VectorOperation::insert);
+    src.import(rw_vector, VectorOperation::insert);
+
+    dst = 0.0;
+  }
+
+  double dt_vmult = 0;
+
+  for (unsigned int i = 0; i < n_repetitions; ++i)
+    {
+      MPI_Barrier(MPI_COMM_WORLD);
+
+      std::chrono::time_point<std::chrono::system_clock> temp =
+        std::chrono::system_clock::now();
+
+      laplace_operator.vmult(dst, src);
+
+      MPI_Barrier(MPI_COMM_WORLD);
+
+      const double dt = std::chrono::duration_cast<std::chrono::nanoseconds>(
+                          std::chrono::system_clock::now() - temp)
+                          .count() /
+                        1e9;
+
+      dt_vmult += dt;
+    }
+
+  dt_vmult = Utilities::MPI::sum(dt_vmult, MPI_COMM_WORLD) /
+             Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD) / n_repetitions;
+
+
+  table.add_value("time_setup", dt_setup);
+  table.set_scientific("time_setup", true);
+  table.add_value("time_avg", dt_vmult);
+  table.set_scientific("time_avg", true);
+
+  if (Utilities::MPI::this_mpi_process(comm) == 0)
+    {
+      table.write_text(std::cout);
+      std::cout << std::endl;
+    }
```

Comment:
It would be best if the performance test driver only outputs the final table. Currently this is interspersed by your convergence table:
```
timing
name    min     max     mean    std_dev samples
n_levels degree n_cells n_dofs  time_setup  time_avg  
       6      4   32768 2146689 2.9011e-02 3.6902e-03 

n_levels degree n_cells n_dofs  time_setup  time_avg  
       6      4   32768 2146689 4.3203e-02 9.3734e-02 

n_levels degree n_cells n_dofs  time_setup  time_avg  
       6      4   32768 2146689 1.6637e-02 4.3178e-03 

n_levels degree n_cells n_dofs  time_setup  time_avg  
       6      4   32768 2146689 5.0174e-02 7.9475e-02 

n_levels degree n_cells n_dofs  time_setup  time_avg  
       6      4   32768 2146689 1.6933e-02 3.5731e-03 

n_levels degree n_cells n_dofs  time_setup  time_avg  
       6      4   32768 2146689 4.0724e-02 8.0566e-02 

n_levels degree n_cells n_dofs  time_setup  time_avg  
       6      4   32768 2146689 1.6584e-02 3.5974e-03 

n_levels degree n_cells n_dofs  time_setup  time_avg  
       6      4   32768 2146689 3.8662e-02 8.3313e-02 

mf_setup        0.0165839       0.0290111       0.0197914       0.0061484       4
mf_vmult        0.00357307      0.00431781      0.00379462      0.000352424     4
mf_kokkos_setup 0.038662        0.0501736       0.0431908       0.00501177      4
mf_kokkos_vmult 0.079475        0.0937337       0.084272        0.00651122      4
```
This breaks the `collect_measurements` script.

Can you guard the table output with a `debug` boolean or something similar?